airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/source.py

```diff
@@ -0,0 +1,407 @@
+        else:
+            data = [json_response]
+
+        for record in data:
+            # In order to support direct datetime string comparison (which is performed in incremental acceptance tests)
+            # convert any date format to python iso format string for date based cursors
+            self.update_field(record, self.cursor_field, lambda date: isoparse(date).isoformat())
+            yield record
+
+        # sleep for 1-2 secs to not reach rate limit: 50 requests per minute
+        time.sleep(60 / self.requests_per_minute)
+
+    @staticmethod
+    def update_field(record: Mapping[str, Any], field_path: Union[List[str], str], update: Callable[[Any], None]):
+
+        if not isinstance(field_path, List):
+            field_path = [field_path]
+
+        last_field = field_path[-1]
+        data = PaypalTransactionStream.get_field(record, field_path[:-1])
+        if data and last_field in data:
+            data[last_field] = update(data[last_field])
+
+    @staticmethod
+    def get_field(record: Mapping[str, Any], field_path: Union[List[str], str]):
+
+        if not isinstance(field_path, List):
+            field_path = [field_path]
+
+        data = record
+        for attr in field_path:
+            if data and isinstance(data, dict):
+                data = data.get(attr)
+            else:
+                return None
+
+        return data
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, any]:
+
+        # This method is called once for each record returned from the API to compare the cursor field value in that record with the current state
+        # we then return an updated state object. If this is the first time we run a sync or no state was passed, current_stream_state will be None.
+        latest_record_date_str: str = self.get_field(latest_record, self.cursor_field)
+
+        if current_stream_state and "date" in current_stream_state and latest_record_date_str:
+            # isoparse supports different formats, like:
+            # python iso format:               2021-06-04T00:00:00+03:00
+            # format from transactions record: 2021-06-04T00:00:00+0300
+            # format from balances record:     2021-06-02T00:00:00Z
+            latest_record_date = isoparse(latest_record_date_str)
+            current_parsed_date = isoparse(current_stream_state["date"])
+
+            return {"date": max(current_parsed_date, latest_record_date).isoformat()}
+        else:
+            return {"date": self.start_date.isoformat()}
+
+
+class Transactions(PaypalTransactionStream):
+    """
+    Stream for Transactions /v1/reporting/transactions
+    """
+
+    data_field = "transaction_details"
+    primary_key = ["transaction_info", "transaction_id"]
+    cursor_field = ["transaction_info", "transaction_initiation_date"]
+
+    start_date_max = {"hours": 36}  # this limit is found experimentally
```

Comment:
I tested different start_date for list transaction and sometimes it worked for start_date = now() - 6 hrs, sometimes only for  start_date = now() - 36 hrs, So I set the biggest value